src/main.rs

```diff
@@ -114,11 +114,40 @@ async fn session_info(
     host_token: String,
     config: &State<Config>,
     db: SessionDBConn,
+    token: TokenCookie,
 ) -> Result<RenderedCredentials, Error> {
-    let credentials = get_credentials_for_host(host_token, config, db)
-        .await
-        .unwrap_or_else(|_| Vec::new());
-    render_credentials(credentials, CredentialRenderType::Json)
+    if check_token(token, config).await? {
+        let credentials = get_credentials_for_host(host_token, config, db)
+            .await
+            .unwrap_or_else(|_| Vec::new());
+        return render_credentials(credentials, CredentialRenderType::Json);
+    }
+
+    Err(Error::Forbidden(
+        "Insufficient rights, try logging in to another account",
+    ))
+}
+
+#[get("/session_info/<host_token>", rank = 2)]
+async fn session_info_anon(host_token: String, config: &State<Config>) -> Result<(), Error> {
+    let login_url = format!("{}/oauth/login?redirect=/logged_in", config.external_url());
+    Err(Error::Unauthorized(login_url))
+}
+
+#[get("/logged_in")]
+async fn logged_in(config: &State<Config>, token: TokenCookie) -> Result<String, Error> {
+    if check_token(token, config).await? {
+        return Ok("You can close this window".to_owned());
+    }
+
+    Err(Error::Forbidden(
+        "Insufficient rights, try logging in to another account",
+    ))
+}
+
+#[get("/logged_in", rank = 2)]
+async fn logged_in_anon() -> Result<String, Error> {
+    Ok("What are you doing here?".to_owned())
```

Comment:
These strings seem to be user visible, so we should make them localizable in my view. Also, a user getting here without being logged in probably signals some sort of error ocurred. The message we return to them should probably reflect that, instead of being some witty frase.